model/proto/api_v2/query.proto

```diff
@@ -0,0 +1,171 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax="proto3";
+
+package jaeger.api_v2;
+
+import "model.proto";
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option go_package = "api_v2";
+option java_package = "io.jaegertracing.api_v2";
+
+// Enable gogoprotobuf extensions (https://github.com/gogo/protobuf/blob/master/extensions.md).
+// Enable custom Marshal method.
+option (gogoproto.marshaler_all) = true;
+// Enable custom Unmarshal method.
+option (gogoproto.unmarshaler_all) = true;
+// Enable custom Size method (Required by Marshal and Unmarshal).
+option (gogoproto.sizer_all) = true;
+// Enable registration with golang/protobuf for the grpc-gateway.
+option (gogoproto.goproto_registration) = true;
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  info: {
+    version: "1.0";
+  };
+  external_docs: {
+    url: "https://github.com/jaegertracing/jaeger";
+    description: "Jaeger API";
+  }
+  schemes: HTTP;
+  schemes: HTTPS;
+};
+
+message GetTraceRequest {
+  bytes trace_id = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+    (gogoproto.customname) = "TraceId"
+  ];
+}
+
+message SpansResponseChunk {
+  repeated jaeger.api_v2.Span spans = 1 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+message ArchiveTraceRequest {
+  bytes trace_id = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.customtype) = "github.com/jaegertracing/jaeger/model.TraceID",
+    (gogoproto.customname) = "TraceId"
+  ];
+}
+
+message ArchiveTraceResponse {
+}
+
+message TraceQueryParameters {
+  string service_name = 1;
+  string operation_name = 2;
+  map<string, string> tags = 3;
+  google.protobuf.Timestamp start_time_min = 4 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false
+  ];
+  google.protobuf.Timestamp start_time_max = 5 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false
+  ];
+  google.protobuf.Duration duration_min = 6 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = false
+  ];
+  google.protobuf.Duration duration_max = 7 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = false
+  ];
+  int32 num_traces = 8;
+}
+
+message FindTracesRequest {
+  TraceQueryParameters query = 1;
+}
+
+message GetServicesRequest {}
+
+message GetServicesResponse {
+  repeated string services = 1;
+}
+
+message GetOperationsRequest {
+  string service = 1;
+}
+
+message GetOperationsResponse {
+  repeated string operations = 1;
+}
+
+message GetDependenciesRequest {
+  google.protobuf.Timestamp start_time = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false
+  ];
+  google.protobuf.Duration lookback = 2 [
+    (gogoproto.stdtime) = true,
```

Comment:
A small mismatch with the plugin API here, as I think `google.protobuf.Duration lookback` better suits `querysvc` APIs. However, can change if required. @yurishkuro

Reply:
I think we intentionally changed it to start_time / end_time, as it has less cognitive overhead (no need to understand yet another type Duration).